Adding django-pytest

rawadabu opened the pull request:
resolve #20 

- Added django-pytest to detect and runs the tests which also provides detailed run report.

NitzanMordhai reviewed (changes requested):
Please change PR description and commit to being more informative, it doesn't say anything about the change.

rawadabu committed:
Adding django-pytest

PyTest is the de-facto standard testing framework in the Python community.
- Detects and runs the tests.
- Provides detailed run report.
- Helps reduce to a minimum boilerplate test code.
- Known for its unique fixtures and generic assertions.

rawadabu commented:
> Please change PR description and commit to being more informative, it doesn't say anything about the change.

Commit description edited and squashed the 2 commits.

Shalevro2 commented:
We need to add also the pytest.yml to .github/workflows directory?
You can find the pytest.yml file at the end of DjangoTestingTutoiral.pdf

rawadabu commented:
> 

Right, created and haven’t added to the branch, thanks for your review @Shalevro2.

rawadabu committed:
Adding pytest.yml

Added pytest.yml file to automatically test our code after creating a PR.

Yarboa commented on pytest.ini:
```diff
@@ -0,0 +1,3 @@
+[pytest]
+DJANGO_SETTINGS_MODULE = finalProject_g4.settings
+python_files = tests.py test_*.py *_tests.py
```
Can you please add a file just make sure that github actions will pass success.
Other wise PR will not move to MERGE
Please refer this 
https://github.com/redhat-beyond/Smarticle/blob/main/CONTRIBUTING.md#guidelines-for-quick-pr-merge

Bullet 5

Verify PR checks all passed with success

rawadabu commented:
So basically PyTest failing for not attaching any test code to the files added to the PR?

Anas-mhameed commented:
Yes I think , Let's add a simple test like : 
def test_example01():
  assert 1 == 1

Yarboa commented on .github/workflows/pytest.yml:
```diff
@@ -0,0 +1,16 @@
+on: pull_request
+jobs:
+  pytest:
+    name: Run the application's tests with PyTest
```
Please consider shorter name
such as:
Run PyTest

rawadabu committed:
Update add-django-pytest branch

1-pytest.yml name changed to Run PyTest
2-Added tests file to make sure that github actions will pass success.

rawadabu committed:
tests file updated

test_django_pytest.py updated due to error running django-pytest.

rawadabu committed:
Updating test_django_pytest.py

Flake8 error fixed.

Shalevro2 reviewed (approved):
LGTM

Anas-mhameed reviewed (approved):
LGTM

finziamit reviewed (approved):
LGTM

yaelnoyman1 reviewed (approved):
LGTM

Yarboa commented:
> Please change PR description and commit to being more informative, it doesn't say anything about the change.

@NitzanMordhai  WDYT? Can you resolve? and merge?

avi-biton commented on pytest.ini:
```diff
@@ -0,0 +1,3 @@
+[pytest]
+DJANGO_SETTINGS_MODULE = finalProject_g4.settings
+python_files = tests.py test_*.py *_tests.py
```
Please add a new line at the end of the file

rawadabu replied:
@avi-biton Done.

Yarboa replied:
LGTM

rawadabu committed:
Updating pytest.ini file

A new line added at the end of the file.
